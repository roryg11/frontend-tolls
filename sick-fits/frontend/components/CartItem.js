import styled from "styled-components";
import formatMoney from "../lib/formatMoney";
import RemoveFromCart from "./RemoveFromCart";

const CartStyles = styled.li`
    padding: 1rem 0;
    display: grid;
    grid-template-columns: auto 1fr auto;
    border-bottom: 1px solid ${ props => props.theme.lightGrey};
    align-items: center;
    img {
        margin-right: 10px;
    }
    h3, p {
        margin: 0; 
    }
`

const CartItem = props => {
    const {cartItem} = props;
    if(!cartItem.item){ 
        return (
            <CartStyles>
                <div>&nbsp;</div>
                <p>This item has been removed</p>
                <RemoveFromCart id={cartItem.id}/>
            </CartStyles>
        )
    };
    return (
        <CartStyles>
            <img width="100" src={cartItem.item.image}/>
            <div className="cart-item-details">
                <h3>{cartItem.item.title}</h3>
                <p>
                    {formatMoney(cartItem.item.price * cartItem.quantity)}
                    {' - '}
                    <em>
                        {cartItem.quantity} &times; {formatMoney(cartItem.item.price)} each
                    </em>
                </p>
            </div>
            <RemoveFromCart id={cartItem.id}/>
        </CartStyles>
    )
}

export default CartItem; 