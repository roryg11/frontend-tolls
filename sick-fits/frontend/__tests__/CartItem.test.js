import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CartItem from "../components/CartItem";

const fakeCartItem = {
    id: "ABC123",
    quantity: 4,
    item: {
        id: "ABF234",
        price: 30505,
        title: "Cutest Planter",
        image: "imAPlant.jpg"
    }

}

describe('<CartItem/>', ()=>{
    it('renders', ()=>{
        shallow(<CartItem cartItem={fakeCartItem}/>);
    });
    it('matches the snapshot', ()=>{
        const cartItemWrapper = shallow(<CartItem cartItem={fakeCartItem}/>);
        expect(cartItemWrapper).toMatchSnapshot(); 
    });
    it('renders the image correctly', ()=>{
        const cartItemWrapper = shallow(<CartItem cartItem={fakeCartItem}/>);
        const img = cartItemWrapper.find('img');
        expect(img.props().src).toEqual(fakeCartItem.item.image);
    });
    it('renders the pricing and quantity and subtotal correctly', ()=>{
        const cartItemWrapper = shallow(<CartItem cartItem={fakeCartItem}/>);
        const pricing = cartItemWrapper.find('p');
        const price = fakeCartItem.item.price * fakeCartItem.quantity;
        expect(pricing.text()).toEqual('$1,220.20 - 4 Ã— $305.05 each');
    });
});