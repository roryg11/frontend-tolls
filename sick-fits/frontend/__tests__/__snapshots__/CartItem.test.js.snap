// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem/> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartItem
    cartItem={
      Object {
        "id": "ABC123",
        "item": Object {
          "id": "ABF234",
          "image": "imAPlant.jpg",
          "price": 30505,
          "title": "Cutest Planter",
        },
        "quantity": 4,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          src="imAPlant.jpg"
          width="100"
        />,
        <div
          className="cart-item-details"
        >
          <h3>
            Cutest Planter
          </h3>
          <p>
            $1,220.20
             - 
            <em>
              4
               × 
              $305.05
               each
            </em>
          </p>
        </div>,
        <RemoveFromCart
          id="ABC123"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "imAPlant.jpg",
          "width": "100",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Cutest Planter
            </h3>,
            <p>
              $1,220.20
               - 
              <em>
                4
                 × 
                $305.05
                 each
              </em>
            </p>,
          ],
          "className": "cart-item-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cutest Planter",
            },
            "ref": null,
            "rendered": "Cutest Planter",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "$1,220.20",
                " - ",
                <em>
                  4
                   × 
                  $305.05
                   each
                </em>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "$1,220.20",
              " - ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    4,
                    " × ",
                    "$305.05",
                    " each",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  4,
                  " × ",
                  "$305.05",
                  " each",
                ],
                "type": "em",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "ABC123",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            src="imAPlant.jpg"
            width="100"
          />,
          <div
            className="cart-item-details"
          >
            <h3>
              Cutest Planter
            </h3>
            <p>
              $1,220.20
               - 
              <em>
                4
                 × 
                $305.05
                 each
              </em>
            </p>
          </div>,
          <RemoveFromCart
            id="ABC123"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "imAPlant.jpg",
            "width": "100",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Cutest Planter
              </h3>,
              <p>
                $1,220.20
                 - 
                <em>
                  4
                   × 
                  $305.05
                   each
                </em>
              </p>,
            ],
            "className": "cart-item-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cutest Planter",
              },
              "ref": null,
              "rendered": "Cutest Planter",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "$1,220.20",
                  " - ",
                  <em>
                    4
                     × 
                    $305.05
                     each
                  </em>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "$1,220.20",
                " - ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      4,
                      " × ",
                      "$305.05",
                      " each",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    4,
                    " × ",
                    "$305.05",
                    " each",
                  ],
                  "type": "em",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "ABC123",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
